test/test_associations.rb

```diff
@@ -228,7 +228,7 @@ def test_double_belongs_to_with_deleted
   end
 
   def test_mass_assignment_of_paranoid_column_enabled
-    if ActiveRecord::VERSION::MAJOR > 4 && ActiveRecord::VERSION::MINOR > 1
+    if Gem.loaded_specs['activerecord'].version >= Gem::Version.new('5.2.0')
```

Comment:
@mvz - there are several places in the code that check the current ActiveRecord version.  Since Rails 6 is out now, it might be worthwhile to release a new major version of this gem where old versions of Rails/Ruby are no longer supported.  A new gem version that supports only Rails 5.2+ w/ Ruby 2.5+ only would be nice clean slate.

Reply:
Yes, that's definitely a good idea!

Reply:
I don't mind working on that after we get this PR merged.